src/pages/settings/settings.js

```diff
@@ -0,0 +1,200 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Form,
+  FormGroup,
+  Grid,
+  GridItem,
+  Card,
+  CardHeader,
+  CardFooter,
+  Button,
+  Page,
+  PageSection,
+  PageSectionVariants,
+  TextArea
+} from '@patternfly/react-core';
+import { withRouter } from 'react-router-dom';
+import { noop } from 'patternfly-react';
+import RoutedConnectedMasthead from '../../components/masthead/masthead';
+import { connect, reduxActions } from '../../redux';
+import Breadcrumb from '../../components/breadcrumb/breadcrumb';
+import { setUserWalkthroughs, getUserWalkthroughs } from '../../services/walkthroughServices';
+
+class SettingsPage extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const { userWalkthroughs } = this.props;
+
+    this.state = {
+      value: userWalkthroughs || '',
+      isValid: true
+    };
+
+    getUserWalkthroughs().then(response => {
+      if (response.data) {
+        this.setState({
+          value: response.data,
+          isValid: true
+        });
+      } else {
+        this.setState({
+          value: '',
+          isValid: true
+        });
+      }
+    });
+  }
+
+  saveSettings = (e, value) => {
+    e.preventDefault();
+    const { history } = this.props;
+    setUserWalkthroughs(value);
+    history.push(`/`);
+  };
+
+  handleTextInputChange = value => {
+    this.setState(
+      {
+        value,
+        isValid: /^(?:https:\/\/)+([w.-]+)+github.com\/[\w\-._~:/?#[\]@!$&/'()*+,;=.]+$/.test(value)
+      },
+      () => {
+        if (this.state.value === '') {
```

Comment:
could value ever be undefined or null?

Reply:
If the field is empty is will be "" and otherwise have some value, which if it isn't a valid github formatted URL, the Save button will be inactive and they won't be able to write it to the db.